components/feature/search/src/main/AndroidManifest.xml

```diff
@@ -2,4 +2,10 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="mozilla.components.feature.search" />
+    xmlns:tools="http://schemas.android.com/tools"
+    package="mozilla.components.feature.search">
+
+    <uses-permission
+        android:name="android.permission.QUERY_ALL_PACKAGES"
```

Comment:
I think I missed this earlier - why do we need the permission to query all packages (apps) installed on the user's phone? This permission is very powerful and requires us to provide a reason on how we use it when submitting an app to the Play store.

The warning `QueryAllPackagesPermission` below is meant to make us aware of what we are requesting, so if we do need this permission here, we should at the very least leave a comment on how it's used. 🙂

Reply:
This is needed for intentSpeech.resolveActivity from 
val intentSpeech = Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH)

        return intentSpeech.resolveActivity(context.packageManager)?.let {
            PendingIntent.getActivity(
                context,
                REQUEST_CODE_VOICE, voiceIntent, PendingIntentUtils.defaultFlags
            )
        }
I can do this in manifest like in Focus and Fenix ,it's the same code or I can add this @SuppressLint("QueryPermissionsNeeded") to the method .

Reply:
This was needed for 
val intentSpeech = Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH)
intentSpeech.resolveActivity(context.packageManager) 

to check if the phone has voice recognition.

Reply:
Left an offline comment to this discuss this since the permission was removed from the change and no longer a concern for this PR.